scripts/partition_manager.py

```diff
@@ -0,0 +1,356 @@
+            continue  # The 'image' being inspected is not an executable, just a place-holder.
+        open(conf['out_path'], 'w').write('''\
+#undef CONFIG_FLASH_BASE_ADDRESS
+#define CONFIG_FLASH_BASE_ADDRESS %s
+#undef CONFIG_FLASH_LOAD_OFFSET
+#define CONFIG_FLASH_LOAD_OFFSET 0
+#undef CONFIG_FLASH_LOAD_SIZE
+#define CONFIG_FLASH_LOAD_SIZE %s
+''' % (hex(conf['address']), hex(conf['size'])))
+
+
+def get_flash_size(config):
+    config.seek(0)  # Ensure that we search the entire file
+    for line in config.readlines():
+        match = re.match(r'#define CONFIG_FLASH_SIZE (\d*)', line)
+        if match:
+            return int(match.group(1)) * 1024
+    raise RuntimeError("Unable to find 'CONFIG_FLASH_SIZE' in any of: %s" % config.name)
+
+
+def generate_override(input_files, output_file_name, configs, app_override_file):
+    adr_map = dict()
+    load_adr_map(adr_map, input_files, output_file_name, app_override_file)
+    load_size_config(adr_map, configs)
+    flash_size = get_flash_size(configs[0])
+    solution, sub_partitions = resolve(adr_map)
+    set_addresses(adr_map, solution, flash_size)
+    set_sub_partition_address_and_size(adr_map, sub_partitions)
+    return adr_map
+
+
+def get_header_guard_start(filename):
+    macro_name = filename.split('.h')[0]
+    return '''/* File generated by %s, do not modify */
+#ifndef %s_H__
+#define %s_H__''' % (__file__, macro_name.upper(), macro_name.upper())
+
+
+def get_header_guard_end(filename):
+    return "#endif /* %s_H__ */" % filename.split('.h')[0].upper()
+
+
+def write_pm_config(adr_map, pm_config_file):
+    lines = list()
+    lines.append(get_header_guard_start(pm_config_file))
+    flash_area_id = 0
+
+    lines.append("\n/* Indirect, iterable list of flash areas */")
+    for area_name, area_props in sorted(adr_map.items(), key=lambda key_value: key_value[1]['address']):
+        lines.append("#define PM_%d_DEV \"NRF_FLASH_DRV_NAME\"" % flash_area_id)
+        lines.append("#define PM_%d_LABEL %s" % (flash_area_id, area_name.upper()))
+        lines.append("#define PM_%d_OFFSET 0x%x" % (flash_area_id, area_props['address']))
+        lines.append("#define PM_%d_SIZE 0x%x" % (flash_area_id, area_props['size']))
+        adr_map[area_name]['flash_area_id'] = flash_area_id
+        flash_area_id += 1
+
+    for area_name, area_props in adr_map.items():
+        lines.append("#define PM_%s_ID %d" % (area_name.upper(), area_props['flash_area_id']))
+    lines.append("#define PM_NUM %d" % flash_area_id)
+
+    lines.append("\n/* Direct look up list of flash areas */")
+    for area_name, area_props in sorted(adr_map.items(), key=lambda key_value: key_value[1]['address']):
+        lines.append("#define PM_%s_ADDRESS 0x%x" % (area_name.upper(), area_props['address']))
+        lines.append("#define PM_%s_SIZE 0x%x" % (area_name.upper(), area_props['size']))
+        lines.append("#define PM_%s_DEV_NAME \"NRF_FLASH_DRV_NAME\"" % area_name.upper())
```

Comment:
NRF_FLASH_DRV_NAME

Any plans for how to make this portable in the future?

Reply:
No, I will investigate why this is needed.